cpp/installer.cc

```diff
@@ -717,20 +723,32 @@ void ConfigureSolrUserAndService(const VuFindSystemType system_type, const bool
 }
 
 
-void SetEnvironmentVariables(const std::string &vufind_system_type_string) {
+void PermanentlySetEnvironmentVariables(std::vector<std::pair<std::string, std::string>> &keys_and_values,
+                                        const std::string &script_path)
+{
+    std::string variables;
+    for (const auto &key_and_value : keys_and_values)
```

Comment:
I think the C++-17 syntax is much nicer: "for (const auto &[key, value] : keys_and_values)" etc.